frontend/features/page/sections/ContactsSection/ContactsSection.tsx

```diff
@@ -0,0 +1,675 @@
+              <Box
+                as="a"
+                fontSize="sm"
+                pt="14"
+                lineHeight="1.8em"
+                textDecoration="none">
+                <InlineTextarea name="email" />
+              </Box>
+            </Link>
+          </Flex>
+          <ContactsForm />
+        </Flex>
+
+        <Flex justifyContent="flex-end" w="full" mb="32">
+          <SocialLink
+            href="https://twitter.com/inkofpixel"
+            color="rgba(29, 161, 242, 0.7)"
+            onHoverColor="rgba(29, 161, 242, 1)">
+            <TwitterIcon color="white" />
+          </SocialLink>
+          <SocialLink
+            href="https://facebook.com/inkofpixel"
+            color="rgba(59, 89, 152, 0.7)"
+            onHoverColor="rgba(59, 89, 152, 1)">
+            <FacebookIcon color="white" />
+          </SocialLink>
+          <SocialLink
+            href="https://github.com/inkofpixel"
+            color="rgba(24, 23, 23, 0.7)"
+            onHoverColor="rgba(24, 23, 23, 1)">
+            <GithubIcon color="white" />
+          </SocialLink>
+        </Flex>
+      </Flex>
+    </Box>
+  );
+}
+
+enum FormStatus {
+  Idle = "idle",
+  Submitting = "submitting",
+  Submitted = "submitted",
+}
+
+const blankForm = {
+  status: FormStatus.Idle,
+  values: { name: "", email: "", message: "" },
+  validationErrors: { name: null, email: null, message: null },
+  submitError: "",
+};
+
+interface UpdateFieldAction {
+  type: FormActionType.UpdateField;
+  name: string;
+  value: string;
+}
+
+interface SuccessAction {
+  type: FormActionType.Success;
+}
+
+interface FailAction {
+  type: FormActionType.Failed;
+}
+
+interface SubmitAction {
+  type: FormActionType.Submit;
+}
+
+enum FormActionType {
+  UpdateField = "update-field",
+  Success = "success",
+  Failed = "failed",
+  Submit = "submit",
+}
+
+type FormAction = UpdateFieldAction | SuccessAction | FailAction | SubmitAction;
+
+interface FormState {
+  status: FormStatus;
+  values: {
+    name: string;
+    email: string;
+    message: string;
+  };
+  validationErrors: {
+    name: Nullable<string>;
+    email: Nullable<string>;
+    message: Nullable<string>;
+  };
+  submitError: string;
+}
+
+function reducer(state: FormState, action: FormAction) {
+  const regex = /^[^@\s]+@[^@\s\.]+\.[^@\.\s]+$/;
+  switch (action.type) {
+    case FormActionType.UpdateField: {
+      return {
+        ...state,
+        values: {
+          ...state.values,
+          [action.name]: action.value,
+        },
+        validationErrors: {
+          ...state.validationErrors,
+          [action.name]: "",
+        },
+        submitError: "",
+      };
+    }
+
+    case FormActionType.Submit: {
+      //Check errors
+
+      let error = false;
+      if (
+        state.values.name == "" ||
+        !state.values ||
+        state.values.name?.replace(/\s/g, "").length == 0
+      ) {
+        state.validationErrors.name = "Please insert your name";
```

Comment:
**Never ever ever ever mutate state** this way inside a reducer. Here you are directly accessing the current state making it effectively mutable. This might produce unintended side effects. You need to store validation errors on a new object, and return the new object from the reducer, so that the original one is unaffected